05week/spaceTravelToMars.js

```diff
@@ -2,14 +2,71 @@
 
 let assert = require('assert');
 
+let good2go = null;
+
 let jobTypes = {
   pilot: 'MAV',
   mechanic: 'Repair Ship',
   commander: 'Main Ship',
   programmer: 'Any Ship!'
 };
 
-// Your code here
+class CrewMember {
+  constructor(name, job, specialSkill){
+    this.name = name;
+    this.job = job;
+    this.specialSkill = specialSkill;
+    this.ship = null;
+  }
+  enterShip(x){
+    this.ship = x;
+    x.crew.push(this);
+  }
+}
+
+class Ship {
+  constructor(name, type, ability){
+    this.name = name;
+    this.type = type;
+    this.ability = ability;
+    this.crew = [];
+  }
+  missionStatement() { //This function returns the mission of the ship if properly crewed and "Can't perform a mission yet." if not
+    console.log("Begin Mission Statement");
+    //console.log("Ship Type: "+(this.type));
+    for (let i=0; i<this.crew.length; i++){
+      let crewMemberJob = this.crew[i].job;
+      console.log("Crew Member Job Type: "+(jobTypes[crewMemberJob]));
+      if (((jobTypes[crewMemberJob])==(this.type))||(this.crew[i].job=='programmer')){
+        good2go = true;
```

Comment:
i would define good2go as a local variable in the function and default it to false.